Source/missiles.cpp

```diff
@@ -200,22 +200,24 @@ bool MonsterMHit(int pnum, int m, int mindam, int maxdam, int dist, missile_id t
 	if (monster._mmode == MonsterMode::Charge)
 		return false;
 
-	uint8_t mor = monster.mMagicRes;
+	MonsterResists mor = monster.mResists;
 	missile_resistance mir = MissilesData[t].mResist;
 
-	if (((mor & IMMUNE_MAGIC) != 0 && mir == MISR_MAGIC)
-	    || ((mor & IMMUNE_FIRE) != 0 && mir == MISR_FIRE)
-	    || ((mor & IMMUNE_LIGHTNING) != 0 && mir == MISR_LIGHTNING)
-	    || ((mor & IMMUNE_ACID) != 0 && mir == MISR_ACID))
+	if ((mor.isMagicImmune() && mir == MISR_MAGIC) || (mor.isFireImmune() && mir == MISR_FIRE) || (mor.isLightningImmune() && mir == MISR_LIGHTNING) || (mor.isAcidImmune() && mir == MISR_ACID)) {
 		return false;
+	}
 
-	if (((mor & RESIST_MAGIC) != 0 && mir == MISR_MAGIC)
-	    || ((mor & RESIST_FIRE) != 0 && mir == MISR_FIRE)
-	    || ((mor & RESIST_LIGHTNING) != 0 && mir == MISR_LIGHTNING))
-		resist = true;
+	if (mor.isMagicResistant() && mir == MISR_MAGIC) {
+		resist = mor.getMagicResist();
+	} else if (mor.isFireResistant() && mir == MISR_FIRE) {
+		resist = mor.getFireResist();
+	} else if (mor.isLightningResistant() && mir == MISR_LIGHTNING) {
+		resist = mor.getLightningResist();
+	}

```

Comment:
I don't see a reason to check before getting?

```suggestion
	resist = mor.getMagicResist(mir);
```

Also, try avoiding acronyms, especially custom ones. I have no idea what mor mean, other them mom in Danish.

Reply:
I assume this is comment would require the element enum comment above to be implemented? I agree that is cleaner, I can implement.

Reply:
Also, even though you edited it out, I think mor was monster resists, and mir was missile resist? I'm not really sure, since they are original, but I can change them to something a bit more descriptive.

Reply:
Yeah, I realized that it wasn't added by you so didn't want to direct it at you. But you are more then welcome to change it to something readable :)